### DISCUSSION TITLE
contributions

### DISCUSSION THREAD

**Discussion by white-room** [nickcoutsos/keymap-editor#207] [General]
Hey @nickcoutsos and contributors -- what a great tool, thanks for your efforts! I've been manually editing config for my Kinesis and just learned about your tool this morning in Ben Frain's video https://www.youtube.com/watch?v=Vy7IoQAe3oU and took it for a spin. Not sure when you stopped updating source code due to corps not wanting to collaborate and build together but do you think you'll reconsider allowing community contributions and collaboration now that ZMK has matured a bit and there are so many options for wireless open source boards? If not publicly is there any limited capacity or private contribution you'd be open to from unaffiliated devs that are just interested in the community? 

Thanks again for a handy and fun tool!

**Comment by astrolemonade** [comment]
Hey! I am also interested in contributing to this specific project. I really consider it has a ton of potential but I understand why is it close source at the moment(to protect the work from companies that use for advertising their stuff without contributing in some way or another to the project).

**Comment by nickcoutsos** [comment]
Firstly thank you both for your interest in this, it means a lot to me to know that people are getting so much out of this and that you would even feel motivated to make contributions of your own is incredible.

I don't think I'll be making moves any time soon to open this up, but Pete is getting underway with ZMK Studio and the functionality to support runtime keymap updates. A while back we had talked about Studio and contributing to it in some way which was exciting but more recently life has thrown me a curveball so it's hard to say how involved I'll get in anything in the near future (my updates to keymap editor have been pretty sparse as well).

That said, I'm still around and although I'm not able to dedicate time to all of the features I've been wanting to implement I'll still listen to ideas and I try to be as responsive as I can to help users with challenges using the app or fixing up show-stopping bugs, so feel free to share ideas and questions!

**Reply by white-room** [reply]
Sounds good, thanks for the update.
